src/build/build.ts

```diff
@@ -108,50 +114,49 @@ export function build(options?: BuildOptions): Promise<any> {
         .pipe(depsProject.split)
         .pipe(optimize(optimizeOptions))
         .pipe(depsProject.rejoin)
-        .pipe(vfs.src(swPrecacheConfig, {
-          cwdbase: true,
-          allowEmpty: true,
-          passthrough: true
-        }));
 
     let allFiles = mergeStream(sourcesStream, depsStream);
 
     let serviceWorkerName = 'service-worker.js';
 
     let unbundledPhase = new ForkedVinylStream(allFiles)
-      .pipe(new SWPreCacheTransform({
-        root,
-        main,
-        buildRoot: 'build/unbundled',
-        serviceWorkerName,
-        configFileName: swPrecacheConfig
-      }))
       .pipe(vfs.dest('build/unbundled'))
 
-    // SWPreCacheTransform needs the deps from bundler after bundles are created
-    // therefore, give transform a promise that is resolved when bundler is done
-    let depsResolve: (value: string[]) => void;
-    let depsPromise = new Promise<string[]>((resolve) => {
-      depsResolve = resolve;
-    });
-
     let bundledPhase = new ForkedVinylStream(allFiles)
       .pipe(bundler.bundle)
-      .on('end', () => {
-        // give the entry points and shared bundle to SWPreCacheTransform
-        let depsList = Array.from(bundler.entrypointFiles.keys());
-        depsList.push(bundler.sharedBundleUrl);
-        depsResolve(depsList);
-      })
-      .pipe(new SWPreCacheTransform({
-        root,
-        main,
-        buildRoot: 'build/bundled',
-        deps: depsPromise,
-        serviceWorkerName,
-        configFileName: swPrecacheConfig
-      }))
       .pipe(vfs.dest('build/bundled'));
+
+    waitForAll([unbundledPhase, bundledPhase]).then(() => {
+      let unbundledDeps = Array.from(bundler.streamResolver.requestedUrls);
+
+      let bundledDeps = Array.from(bundler.entrypointFiles.keys());
+      bundledDeps.push(bundler.sharedBundleUrl);
+
+      parsePreCacheConfig(swPrecacheConfig).then((swConfig) => {
+        let unbundledConfig = clone(swConfig);
+        let bundledConfig = clone(swConfig);
+        Promise.all([
+          generateServiceWorker({
```

Comment:
how about making a local wrapper on `generateServiceWorker` that takes care of the common params?